src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -137,6 +140,67 @@ private string ResolveMatchedPath(string matchedPath)
             return RequestPathBase.Add(relativePath).ToString();
         }
 
+        private class PathComparer : IComparer<string>
+        {
+            public int Compare(string x, string y)
+            {
+                // < 0 = x < y
+                // > 0 = x > y
+
+                if (x.Equals(y))
+                {
+                    return 0;
+                }
+
+                var xExtIndex = x.LastIndexOf('.');
+                var yExtIndex = y.LastIndexOf('.');
+
+                // Ensure extension index is in the last segment, i.e. in the file name
+                var xSlashIndex = x.LastIndexOf('/');
+                var ySlashIndex = y.LastIndexOf('/');
+                xExtIndex = xExtIndex > xSlashIndex ? xExtIndex : -1;
+                yExtIndex = yExtIndex > ySlashIndex ? yExtIndex : -1;
+
+                // Get paths without their extensions, if they have one
+                var xNoExt = xExtIndex >= 0 ? x.Substring(0, xExtIndex) : x;
```

Comment:
Might want to do a `Debug.Assert` `xExtIndex >= 0` before this. Only thing ensuring that it is is in your `ResolveMatchedPath` bit which prepends a `/` to the path.